src/core/parser.ts

```diff
@@ -35,13 +36,73 @@ const formatExtensions: Record<DiagnosticsFormat, string> = {
 
 const validFormats = ['stylish', 'json', 'junit', 'html', 'text', 'teamcity', 'pretty'];
 
+type AuthEntry = {
+  pattern: string;
+  token: string;
+};
+
+const CONFIG_FILE = path.join(os.homedir(), '.asyncapi', 'config.json');
+
+function loadAuthEntries(): AuthEntry[] {
+  if (!fs.existsSync(CONFIG_FILE)) { return [];}
+  const content = fs.readFileSync(CONFIG_FILE, 'utf8');
+  try {
+    const parsed = JSON.parse(content);
+    return Array.isArray(parsed?.auth) ? parsed.auth : [];
+  } catch (e) {
+    console.error('❌ Invalid JSON in config file:', e);
+    return [];
+  }
+}
+
+function normalizeRawGitUrl(uri: URI): string {
+  // Converts: raw.githubusercontent.com/org/repo/branch/path → https://github.com/org/repo/tree/branch
+  const segments = uri.path().split('/').filter(Boolean); // Remove empty segments
+  if (segments.length < 3) { return uri.toString(); }
+
+  const [org, repo, branch] = segments;
+  return `https://github.com/${org}/${repo}/tree/${branch}`;
+}
+
+const customResolver = {
+  schema: 'https',
+  order: 1,
+
+  canRead: (uri: URI) => uri.hostname() === 'raw.githubusercontent.com',
+
+  read: async (uri: URI) => {
+    const url = uri.toString();
+    const normalized = normalizeRawGitUrl(uri);
```

Comment:
What if the URL is not pointing to a GitHub repo? This could perfectly be a Schema Registry URL, an S3 bucket, some random URL, or even a GitLab or BitBucket URL.

Reply:
Just to be sure we're understanding each other. If I'm understanding it correctly, right now we're assuming all URLs are GitHub URLs, which is wrong because that would make parser stop working with custom URLs. In the `normalizeRawGitUrl` function we should be checking if it's actually a GitHub URL, otherwise, leave it intact. Mappings for GitLab and Bitbucket can wait but breaking all URLs except the GitHub ones is definitely not acceptable.

Reply:
yes, i need to change the functionnaming properly i will make it generic so that we can normalise other url as well in the future based on the requirements